### PR TITLE
docs: add keyauth doc

### PR DESCRIPTION
**Pull request by li-jin-gou**
#### What type of PR is this?

doc

#### What this PR does / why we need it (en: English/zh: Chinese):

en: add keyauth doc

### CODE REVIEW

**Comment by li-jin-gou** [comment]
Welcome CR and not merged for now

**Comment by li-jin-gou** [comment]
参考 https://github.com/cloudwego/cloudwego.github.io/pull/365 完善 example 示例代码保证 简单示例的main函数是全的

File: content/zh/docs/hertz/tutorials/basic-feature/middleware/keyauth.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
+---
+title: "KeyAuth"
+date: 2022-09-22
+weight: 7
+description: >
+
+---
+
+Hertz 提供了 [keyuath](https://github.com/hertz-contrib/keyauth) 扩展用于帮助用户实现 `token` 鉴权。 [keyuath](https://github.com/hertz-contrib/keyauth) 扩展的实现参考了 [Fiber](https://github.com/gofiber/fiber) 和 [Echo](https://github.com/labstack/echo) 的实现。
```
Comment (Skyenought) [inline content/zh/docs/hertz/tutorials/basic-feature/middleware/keyauth.md:9]: typo

Comment (li-jin-gou) [reply content/zh/docs/hertz/tutorials/basic-feature/middleware/keyauth.md:9]: thanks

File: content/en/docs/hertz/tutorials/basic-feature/middleware/keyauth.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+	"github.com/hertz-contrib/keyauth"
+)
+
+func main() {
+	h := server.Default()
+	h.Use(keyauth.New(
+		keyauth.WithContextKey("token"),
+		keyauth.WithKeyLookUp("query:token", ""),
+	))
+	h.GET("/ping", func(c context.Context, ctx *app.RequestContext) {
```
Comment (rogerogers) [inline content/en/docs/hertz/tutorials/basic-feature/middleware/keyauth.md:38]: 这个我看其他example倾向于前面是ctx后面是c。

File: content/en/docs/hertz/tutorials/basic-feature/middleware/keyauth.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+		ctx.JSON(consts.StatusOK, utils.H{"ping": value})
+	})
+	h.Spin()
+}
+```
+
+### WithErrorHandler
+
+`keyauth` 扩展提供了 `WithErrorHandler` 用于帮助用户设置校验 `token` 失败的自定义处理逻辑 。
+
```
Comment (L2ncE) [inline content/en/docs/hertz/tutorials/basic-feature/middleware/keyauth.md:166]: 这里应该是英文吧

Comment (li-jin-gou) [reply content/en/docs/hertz/tutorials/basic-feature/middleware/keyauth.md:166]: thanks

**Review by GuangmingLuo — APPROVED** [review]

**Pull request merged**